File: bin/bin/old_bin/bin/math_0003.c
#include <stdio.h>
#include <stdbool.h>

const int TARGET=1000;

void main(void){
	int v_from=-TARGET;
	int v_to=TARGET;
	int a, b, c;
	int count=0;
	for (a=v_from; a <= v_to; a++){
	for (b=v_from; b <= v_to; b++){
	for (c=v_from; c <= v_to; c++){
		int test=(a*b*c);
		bool valid = (test == TARGET);
		if (valid){
			printf("a=%d, b=%d, c=%d\n", a, b, c);
			count++;
		}
	}
	}
	}
	printf("Number of solution is: %d\n", count);
}

File: bin/bin/old_bin/math_0003.py
#!/usr/bin/python
import math
#How many ordered integer (a,b,c) solutions does the equation abc = 1000 have? How quickly can you count them

class exercise :
    target=1000
    def doit(self):
	v_from=(-1*self.target)+1
	v_to=self.target-1
	result=self.target
        count=0
        for a in range(v_from, v_to):
            for b in range(v_from, v_to):
		for c in range(v_from, v_to):
	                test = (a * b * c)
        	        valid = (test == result)
			count += valid 
			if (valid):
				print("a = {}, b={}, c={}").format(a,b,c)
        print('count is {}').format(count)

	
if __name__ == '__main__':	
	x = exercise()
	x.doit()

